File: app/src/main/java/com/lamnt/foodorder/utils/ValidateUtil.java
package com.lamnt.foodorder.utils;

import android.widget.EditText;

import java.util.HashMap;
import java.util.List;

public class ValidateUtil {
    public static boolean checkNull(Object input) {
        if (input == null) {
            return true;
        }
        if (input instanceof String) {
            return input.toString().trim().isEmpty();
        }
        if (input instanceof EditText) {
            return ((EditText) input).getText().toString().trim().isEmpty();
        }
        if (input instanceof List) {
            return ((List) input).isEmpty();
        }

        if (input instanceof HashMap) {
            return ((HashMap) input).isEmpty();
        }

        return false;
    }
}

File: app/src/main/java/com/lamnt/foodorder/utils/ValidateUtil.kt
package com.lamnt.foodorder.utils

import android.widget.EditText
import java.util.*

object ValidateUtil {
    @JvmStatic
    fun checkNull(input: Any?): Boolean {
        if (input == null) {
            return true
        }
        if (input is String) {
            return input.toString().trim { it <= ' ' }.isEmpty()
        }
        if (input is EditText) {
            return input.text.toString().trim { it <= ' ' }.isEmpty()
        }
        if (input is List<*>) {
            return input.isEmpty()
        }
        return if (input is HashMap<*, *>) {
            input.isEmpty()
        } else false
    }
}